### ISSUE TITLE
dispatcher get request as 'str' only

### ISSUE THREAD

**Issue by yakkle** [bcb/jsonrpcserver#130]
If I call dispatcher with request as bytes, that doesn't work.
```py
    def dispatch():
         request_bytes = b'{"id": 1, "jsonrpc": "2.0",  "method": "get_name"}'
         await async_dispatch(request_bytes, methods, context=context)
```

**Comment by bcb** [comment]
Yes many of the libraries require you to decode the request before passing it to jsonrpcserver, ([examples](https://jsonrpcserver.readthedocs.io/en/latest/examples.html))

Isn’t it best for the implementation to do it, not jsonrpcserver? Jsonrpcserver doesn’t know how to decode it.
